larray/viewer.py

```diff
@@ -1851,8 +1859,150 @@ def __init__(self, parent=None):
         self.data = None
         self.arraywidget = None
 
-    def setup_and_check(self, data, title='', readonly=False,
-                        xlabels=None, ylabels=None,
+        self.setup_menu_bar()
+
+    def _setup_and_check(self, data, title, readonly):
+        """Setup editor"""
+        self.data = data
+
+        icon = self.style().standardIcon(QStyle.SP_ComputerIcon)
+        if icon is not None:
+            self.setWindowIcon(icon)
+
+        if not title:
+            title = 'Session' if self.__class__ == MappingEditor else 'Array'
+            title = _(title + " viewer") if readonly else _(title + " editor")
+        if readonly:
+            title += ' (' + _('read only') + ')'
+        self.setWindowTitle(title)
+
+        widget = QWidget()
+        self.setCentralWidget(widget)
+
+        layout = QVBoxLayout()
+        widget.setLayout(layout)
+
+        return layout
+
+    def setup_menu_bar(self):
+        """Setup menu bar"""
+        menu_bar = self.menuBar()
+        file_menu = menu_bar.addMenu('File')
+
+        file_menu.addAction(create_action(self, _('New'), shortcut=QKeySequence.New,
+                                          icon=None, triggered=self.new))
+        file_menu.addAction(create_action(self, _('Open'), shortcut=QKeySequence.Open,
+                                          icon=None, triggered=self.open,
+                                          statustip=_('Load session from file')))
+        file_menu.addAction(create_action(self, _('Save'), shortcut=QKeySequence.Save,
+                                          icon=None, triggered=self.save,
+                                          statustip=_('Save all arrays and executed commands in new directory')))
+        file_menu.addAction(create_action(self, _('Save Data'), shortcut=QKeySequence("Shift+S"),
+                                          icon=None, triggered=self.saveData,
+                                          statustip=_('Save all current arrays in a Session file')))
+        file_menu.addAction(create_action(self, _('Save As'), shortcut=QKeySequence.SaveAs,
```

Comment:
None of these keyboard shortcuts seem to work currently. Unsure how to fix that nicely, but hardcoding the shortcuts as QShortcut like I did in `ArrayView.__init__` seems to work. There ought to be a better way.